### PR TITLE
#223: Added create checkout with previously selected addresses

### PR DESCRIPTION
**Pull request by ahmed-shehata**
#223: Added create checkout with previously selected addresses

### CODE REVIEW

File: AtlasUI/AtlasUI/AtlasCheckout.swift [hunk 1]
```diff
@@ -74,22 +74,25 @@ public class AtlasCheckout: LocalizerProviderType {
...
-                    let checkoutModel = CheckoutViewModel(article: article,
-                        selectedUnitIndex: selectedUnitIndex, cartId: cartId, checkout: nil)
-                    completion(.success(checkoutModel))
-                }
-
-            case .success(let checkout):
-                let checkoutModel = CheckoutViewModel(article: article,
-                    selectedUnitIndex: selectedUnitIndex, checkout: checkout)
-                completion(.success(checkoutModel))
+    func createCheckoutViewModel(withArticle article: Article, selectedUnitIndex: Int, checkoutViewModel: CheckoutViewModel? = nil,
```
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/AtlasCheckout.swift]: `createCheckoutViewModel` takes `CheckoutViewModel` ? 😟

Comment (ahmed-shehata) [inline AtlasUI/AtlasUI/AtlasCheckout.swift]: Optional one, to update if possible (billing/shipping/..)

Comment (bonkey) [inline AtlasUI/AtlasUI/AtlasCheckout.swift]: So maybe "prepare" instead "create". Don't lie in code, because it will lie to you one day.

Comment (bonkey) [inline AtlasUI/AtlasUI/AtlasCheckout.swift]: An actually why we can't use "updateCart" for this? 

I mean I can guess why, but you should ask yourself and maybe create another method, or create a common function, and use them in both: create, update and so on.

One method must do just one thing.

**Comment by hani-ibrahim** [comment]
👍

**Comment by artdaw** [comment]
👍

**Pull request merged**